src/blocker.cr

```diff
@@ -52,10 +53,18 @@ class Castblock::Blocker
 
   private def watch_device(device : Chromecast::Device, continue : Channel(Nil)) : Nil
     @chromecast.start_watcher(device, continue) do |message|
-      Log.debug &.emit("Received message", application: message.application.display_name)
+      if application = message.application
+        Log.debug &.emit("Received message", application: application.display_name)
+        if application.display_name.downcase == "youtube" && (media = message.media) && media.player_state == "PLAYING"
+          handle_media(device, media)
+        end
+      end
 
-      if message.application.display_name.downcase == "youtube" && (media = message.media) && media.player_state == "PLAYING"
-        handle_media(device, media)
+      if @mute_ads && (payload = message.payload)
+        data = JSON.parse(payload)
+        if data["status"]? && (player_state = data["status"][0]["customData"]["playerState"].as_i)
+          handle_monetization(device, player_state)
```

Comment:
By any chance can we know with the `message` object if the player is already muted? if this needs another request then it is useless to check.

Reply:
Yep this could be improved, I haven't added a check for the current mute status. It seems the payload has the mute status, so that could reduce redundant calls to mute/unmute.